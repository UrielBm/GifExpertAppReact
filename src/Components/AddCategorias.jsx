import React, {useState } from 'react'
import PropTypes from 'prop-types';
export const AddCategorias = ({setCategorias}) => {
    const [InputValue, setInputValue] = useState('');
    const OnChangeValue= (e) =>{
        setInputValue(e.target.value);
        console.log('Onchange llamado')
    };
    const handleSumit= (e)=>{
        e.preventDefault();
        if(InputValue.trim().length>2){
        setCategorias(categ=>[InputValue,...categ,]);
        setInputValue('')
        };
    };
    return (
        <form className="wrapper-form" onSubmit={handleSumit}>
            <p>{InputValue}</p>
            <label>
                <input type="text"  className="input-categoria" placeholder='"ingresa tu  nueva categoria"' value={InputValue} onChange={OnChangeValue} /> 
            </label>
        </form>
    );
}
AddCategorias.propTypes = {
    setCategorias: PropTypes.func.isRequired
};
