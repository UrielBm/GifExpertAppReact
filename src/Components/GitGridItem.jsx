import React from 'react'
import PropTypes from 'prop-types';
export const GitGridItem = ({id,title,url}) => {
    return (
        <div className="wrapper-gift-item">
            {/* {imagenes.title} */}
            < img  className="gift" src={url} alt={title} />
            <p className="titulo-gift animate__animated animate__bounceIn animate__delay-1s ">{title}</p>
        </div>
    )
}
GitGridItem.propTypes={
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}