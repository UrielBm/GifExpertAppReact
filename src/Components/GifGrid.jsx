import React, {Fragment}from 'react'
import { useFectchGifts } from '../hooks/useFectchGifs'
 import { GitGridItem } from './GitGridItem';
export const GifGrid = ({categoria}) => {
     const {loading,data:imagenes}=useFectchGifts(categoria);
    // console.log(data);

    return (
        <Fragment>
            <h3 className="titulo-Categoria">{categoria}</h3>
            {loading&& <p className="animate__animated animate__flash">Cargando...</p>}
            <div className="wrapper-card-gift-item animate__animated animate__bounceInDown">
                {
                    imagenes.map((imagenes)=>(
                        <GitGridItem key={imagenes.id} {...imagenes} />
                    ))
                }
                
            </div>
        </Fragment>
    )
}
