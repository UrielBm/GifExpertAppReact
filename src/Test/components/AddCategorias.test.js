import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { AddCategorias } from '../../Components/AddCategorias';
describe('Evaluando el componente AddCategorias.js', () => {
    const setCategorias=jest.fn();
    let wrapper=shallow(<AddCategorias setCategorias={setCategorias} />)
    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper=shallow(<AddCategorias setCategorias={setCategorias} />)
    })
    test('evaluando que se visualize bien el componente ', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('debe cambios en un input ', () => {
        const input=wrapper.find('input');
        const value='hola mundo;'
        input.simulate('change',{target:{ value:value }});
         const parrafoinput=wrapper.find('p').text().trim();
        expect(parrafoinput).toBe(value);
    });
    test('NO DEBE DE POSTEAR INFORMACION DEL SUBMIT ', () => {
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategorias).not.toHaveBeenCalled();
    });
    test('llamado de setCategorias y limpiar la caja de texto', () => {
        const value='hola mundo';
        wrapper.find('input').simulate('change',{target:{value}});
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategorias).toHaveBeenCalledWith(expect.any(Function));
        expect(wrapper.find('input').prop('value')).toBe('');

    })
    
    
})
