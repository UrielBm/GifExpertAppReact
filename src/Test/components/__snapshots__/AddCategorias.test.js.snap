// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluando el componente AddCategorias.js evaluando que se visualize bien el componente  1`] = `
<form
  className="wrapper-form"
  onSubmit={[Function]}
>
  <p />
  <label>
    <input
      className="input-categoria"
      onChange={[Function]}
      placeholder="\\"ingresa tu  nueva categoria\\""
      type="text"
      value=""
    />
  </label>
</form>
`;
